File: Part1_강의자료/Ch08_투포인터/문제별코드/2230_수고르기/2230.cpp
#include <stdio.h>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int N, M;

    scanf("%d %d", &N, &M);
    vector<int> v(N);
    for (int i = 0; i < N; i++)
        scanf("%d", &v[i]);

    sort(v.begin(), v.end());

    int ans_diff = v[N - 1] - v[0];
    int right = 0;
    for (int left = 0; left < N; left++)
    {
        while (v[right] - v[left] < M && right < N - 1)
            right++;
        int diff = v[right] - v[left];
        if (diff >= M)
            ans_diff = min(ans_diff, diff);
    }

    printf("%d\n", ans_diff);
    return 0;
}

File: Part1_강의자료/Ch08_투포인터/문제별코드/2230_수고르기/2230.py
N, M = map(int, input().split())
arr = [int(input()) for _ in range(N)]

arr.sort()

ans_diff = arr[N - 1] - arr[0]
right = 0
for left in range(N):
    while right < N - 1 and arr[right] - arr[left] < M:
        right += 1
    diff = arr[right] - arr[left]
    if diff >= M:
        ans_diff = min(ans_diff, diff)

print(ans_diff)
